### PR TITLE
configurable overlay line colors

### LINKED ISSUE #3657: crop and rotate: colored guides
**Is your feature request related to a problem? Please describe.**
The guide lines in grey colors are sometimes hard to see, especially when working monochrome. Example:
crop_guide01

**Describe the solution you'd like**
With a color picker for bright and dark color the user can choose his own colors for the guide lines. Example:
crop_guide02

### PR DESCRIPTION
**Pull request by elstoc**
Adds a new bottom-panel darkroom icon to allow user to choose the color of the lines
that display in the following scenarios:

 - when creating or editing masks (including in spot removal and retouch
   module)
 - in the vignetting module
 - in the graduated density module
 - in the crop and rotate module (guide lines and keystone correction)

color options available are gray (default), red, green, yellow, cyan, magenta.

A keyboard shortcut is available to cycle through the available options (default ctrl-o).

Resolves #3657

### CODE REVIEW

**Comment by TurboGit** [comment]
I was planning on working on this :)

Anyway, please add a key shortcut to dynamically change the overlay color while editing a form, it would be a shame to have to go to perfs for this all the time.

Currently O and Maj+O are used to activate the overexposed zones and raw overexposed zones, I would use Ctrl+O maybe (as O for overlay) ?

**Comment by elstoc** [comment]
Ok I'll take a look at this tomorrow.

**Comment by elstoc** [comment]
I've moved the setting to the bottom bar in the darkroom but still a couple of things to resolve:

- I don't have an icon so I reused the overexposed indicator icon
- The popup is a bit wide and I don't know how to make it smaller

Other than that all seems to work as it should. Thanks to @houz for the assistance.

**Comment by elstoc** [comment]
Have now added an icon and resized the popup. This should be complete now, unless anyone would like me to add some more colors to the list. I've tried to only include colors that might stand out against the background.

**Comment by elstoc** [comment]
I wonder whether the crop & rotate guide lines being brighter (or more opaque) might also help with visibility?

File: src/views/darkroom.c [hunk 1]
```diff
@@ -2239,6 +2282,47 @@ void gui_init(dt_view_t *self)
...
+                     G_CALLBACK(_overlay_color_quickbutton_clicked), dev);
+    g_signal_connect(G_OBJECT(dev->overlay_color.button), "button-press-event",
+                     G_CALLBACK(_overlay_color_quickbutton_pressed), dev);
+    g_signal_connect(G_OBJECT(dev->overlay_color.button), "button-release-event",
+                     G_CALLBACK(_overlay_color_quickbutton_released), dev);
+    dt_view_manager_module_toolbox_add(darktable.view_manager, dev->overlay_color.button, DT_VIEW_DARKROOM);
+
+    // and the popup window
+    dev->overlay_color.floating_window = gtk_popover_new(dev->overlay_color.button);
+    gtk_widget_set_size_request(GTK_WIDGET(dev->overlay_color.floating_window), dialog_width/2.0, -1);
```
Comment (TurboGit) [inline src/views/darkroom.c:2302]: The / 2.0 here is a bit too much, the magenta string is long and gets close to the label. We need a bit more width, especially for some translated languages like French where "overlay color" will probably be "couleur des superpositions" ou "couleur des contrôles".

**Comment by TurboGit** [comment]
> I wonder whether the crop & rotate guide lines being brighter (or more opaque) might also help with visibility?

Will see, but this would be in another PR.

File: src/views/darkroom.c (hunk 1 above)
Comment (elstoc) [reply src/views/darkroom.c:2302]: What would you recommend? Should I just leave it at default (dialog_width)?

Comment (TurboGit) [reply src/views/darkroom.c:2302]: Yes probably as a first approach. I'm not sure using some magic number like 1.2f or 1.3f will be better.

Comment (elstoc) [reply src/views/darkroom.c:2302]: ok will do

Comment (elstoc) [reply src/views/darkroom.c]: done

**Comment by TurboGit** [comment]
Good to me now. Thanks for this nice addition.

**Comment by TurboGit** [comment]
BTW, did you try using different colors for each line in the little icons to convey the fact that it is about colors?

**Comment by elstoc** [comment]
No I didn't. I was quite pleased that I worked out how to create the icon at all!

**Comment by elstoc** [comment]
Happy to give it a shot if you like.

**Comment by TurboGit** [comment]
Yes please, if you have some spare cycle :)

**Comment by elstoc** [comment]
No probs. I'm running out of things to do.

**Comment by AxelG-DE** [comment]
I am happy about this new function
Just one question....
Why red and magenta are solid lines, where all the others are dotted lines?

**Comment by elstoc** [comment]
They're not solid. The difference between light and dark is just less pronounced when viewed against a light background. Try viewing them against a dark background.

**Comment by AxelG-DE** [comment]
I see, what you mean, the space in between the dots are dark gray. would it be helpful to have that part transparant? Because it really looks solid :-)
Screenshot_20200406_213324

**Comment by elstoc** [comment]
I believe the reason it alternates between solid and light color is so that the gridlines show against both dark and light backgrounds. If you made the light bit transparent it might not show up against dark backgrounds. Similarly if you made the dark bit transparent it might not show up against light backgrounds. Some colors work better than others.

**Comment by AxelG-DE** [comment]
agree on this comment :)

**Pull request merged**